src/OpenTelemetry/Metrics/AggregatorStore.cs

```diff
@@ -124,17 +128,17 @@ internal sealed class AggregatorStore
 
         if (this.OutputDeltaWithUnusedMetricPointReclaimEnabled)
         {
-            this.availableMetricPoints = new Queue<int>(cardinalityLimit - reservedMetricPointsCount);
+            this.availableMetricPoints = new Queue<int>(this.NumberOfReservedMetricPoints - reservedMetricPointsCount);
```

Comment:
This should simply be the user provided cardinality now.

```suggestion
            this.availableMetricPoints = new Queue<int>(cardinalityLimit);
```

Reply:
Both are same. I agree cardinalityLimit is the better fit here.